scripts/src/public/services/native_offers.ts

```diff
@@ -40,22 +47,36 @@ export function isPayToUser(jwt: ExternalOrderJWT): jwt is ExternalPayToUserOrde
 	return jwt.sub === "pay_to_user";
 }
 
-export async function validateExternalOrderJWT(jwt: string, appUserId: string, logger: LoggerInstance): Promise<ExternalOrderJWT> {
+export async function validateExternalOrderJWT(jwt: string, user: User, logger: LoggerInstance): Promise<ExternalOrderJWT> {
 	const decoded = await verifyJWT<PayToUserPayload, "spend" | "earn" | "pay_to_user">(jwt, logger);
 
 	if (decoded.payload.sub !== "earn" && decoded.payload.sub !== "spend" && decoded.payload.sub !== "pay_to_user") {
 		throw InvalidExternalOrderJwt();
 	}
 
-	if ((decoded.payload.sub === "spend" || decoded.payload.sub === "pay_to_user") &&
-		!!decoded.payload.sender.user_id && decoded.payload.sender.user_id !== appUserId) {
+	// TEMP:JID_MIGRATION
+	if (decoded.payload.iss === "kik") {
+		const userAppId = user.appUserJid ? user.appUserId : null;
+		const userAppJid = user.appUserJid ? user.appUserJid : user.appUserId;
+		const decodedUserAppId = decoded.payload.sender.user_jid ? decoded.payload.sender.user_id : null;
+		const decodedUserAppJid = decoded.payload.sender.user_jid ? decoded.payload.sender.user_jid : decoded.payload.sender.user_id;
+
+		if ((decoded.payload.sub === "spend" || decoded.payload.sub === "pay_to_user")) {
```

Comment:
this might fail if kik only changes the register JWT and not the spend. so `userAppId` will be defined but `decodedUserAppId` won't. Should you support this case?

Reply:
but if they don't change their spend JWT then they will send the `jid` as the regular id, so the matching will fail in any case...
they have to change all JWTs or none at all